main.tf

```
@@ -75,10 +77,13 @@ EOF
 resource "aws_s3_bucket" "quortex" {
   for_each = local.buckets
 
-  bucket        = "${var.storage_prefix}-${each.value}"
+  bucket        = "${var.storage_prefix}-${each.key}"
   force_destroy = var.force_destroy
 
-  tags = var.tags
+  tags = merge(
+    each.value,
+    var.tags
+  )
```

I don't know if this is supposed to happen but bucket specific tags should override general tags.
```suggestion
  tags = merge(
    var.tags,
    each.value
  )
```